anta/tests/snmp.py

```diff
@@ -237,3 +240,129 @@ def test(self) -> None:
             self.result.is_failure(f"Expected `{self.inputs.contact}` as the contact, but found `{contact}` instead.")
         else:
             self.result.is_success()
+
+
+class VerifySNMPNotificationHost(AntaTest):
+    """Verifies the SNMP notification host (SNMP manager) configurations.
+
+    - Verifies that the valid notification type and VRF name.
+    - Ensures that UDP port provided matches the expected value.
+    - Ensures that the community_string is properly set for SNMP v1/v2 and for SNMP v3, the user field is included, aligning with version-specific requirements.
+
+    Expected Results
+    ----------------
+    * Success: The test will pass if the provided SNMP notification host and all specified parameters are correctly configured.
+    * Failure: The test will fail if the provided SNMP notification host is not configured or specified parameters are not correctly configured.
+
+    Examples
+    --------
+    ```yaml
+    anta.tests.snmp:
+      - VerifySNMPNotificationHost:
+          notification_hosts:
+            - hostname: 192.168.1.100
+              vrf: default
+              notification_type: trap
+              version: v1
+              udp_port: 162
+              community_string: public
+              user: public
+    ```
+    """
+
+    name = "VerifySNMPNotificationHost"
+    description = "Verifies the SNMP notification host (SNMP manager) configurations."
+    categories: ClassVar[list[str]] = ["snmp"]
+    commands: ClassVar[list[AntaCommand | AntaTemplate]] = [AntaCommand(command="show snmp notification host", revision=1)]
+
+    class Input(AntaTest.Input):
+        """Input model for the VerifySNMPNotificationHost test."""
+
+        notification_hosts: list[SNMPHost]
+        """List of SNMP hosts."""
+
+        class SNMPHost(BaseModel):
+            """Model for a SNMP Manager."""
+
+            hostname: IPv4Address
+            """IPv4 address of the SNMP notification host."""
+            vrf: str = "default"
+            """Optional VRF for SNMP Hosts. If not provided, it defaults to `default`."""
+            notification_type: Literal["trap", "inform"]
+            """Type of SNMP notification (trap or inform)."""
+            version: Literal["v1", "v2c", "v3"]
```

Comment:
i think the snmp version you can move to custom type as we can use it for many other general snmp testcases

Reply:
Added to custom types.